plugins/generator-1.18.2/forge-1.18.2/templates/elementinits/villagertrades.java.ftl

```diff
@@ -61,15 +61,14 @@ import net.minecraft.world.entity.npc.VillagerTrades;
     }
     </#if>
 
-    <#if trade.hasNormalTrade()>
+    <#if w.hasVillagerTrade(true)>
```

Comment:
This will not work unless you add the following method to `WorkspaceInfo`:
```java
	public boolean hasVillagerTradesOfType(String type) {
		for (ModElement element : workspace.getModElements())
			if (element.getType() == ModElementType.VILLAGERTRADE) {
				if (element.getGeneratableElement() instanceof VillagerTrade vt)
					if (vt.hasVillagerTrade(wandering))
						return true;
			}
		return false;
	}
```

Reply:
Which one? There are 2 of them

Reply:
Also your code won't work because `wandering` is used but not defined

Reply:
Oh, sorry about that.
>Which one? There are 2 of them

It should be in the plural like other WorkspaceInfo methods.